resolve/utilities/params/CMakeLists.txt

```
@@ -0,0 +1,27 @@
+#[[
+
+@brief Build ReSolve function that returns version at runtime
+
+@author Slaven Peles <peless@ornl.gov>
+
+]]
+
+set(Options_SRC 
+  CliOptions.cpp
+)
+
+set(Options_HEADER_INSTALL
+  CliOptions.hpp
+)
+
+# Build shared library ReSolve
+add_library(resolve_options OBJECT ${Options_SRC})
+set_property(TARGET resolve_options PROPERTY POSITION_INDEPENDENT_CODE ON)
+
+target_include_directories(resolve_options PUBLIC
+    $<BUILD_INTERFACE:${PROJECT_SOURCE_DIR}>
+    $<BUILD_INTERFACE:${CMAKE_BINARY_DIR}>
+    $<INSTALL_INTERFACE:include>
+)
+
+install(FILES ${Options_HEADER_INSTALL} DESTINATION include/resolve/utilities/options)
```

```suggestion
install(FILES ${Options_HEADER_INSTALL} DESTINATION include/resolve/utilities/options)

target_link_libraries(...)
```

Per my previous comments, we would just be moving where the `target_link_libraries` call actually happens. Again, purely a cosmetic suggestion

---

We still need two different calls for linking libraries. One is for public linking of shared libraries and the other one is for private linking of object libraries.

---

I think we should refactor CMake per your suggestion. That would make configuration easier to understand. See #145